ballerina/client.bal

```diff
@@ -341,6 +341,192 @@ public isolated client class Client {
+        if cannedACL is CannedACL {
+            requestHeaders[X_AMZ_ACL] = cannedACL;
+        }
+        populateMultipartUploadHeaders(requestHeaders, multipartUploadHeaders);
+
+        check generateSignature(self.accessKeyId, self.secretAccessKey, self.region, POST, requestURI, UNSIGNED_PAYLOAD,
+            requestHeaders, request, queryParams = {"uploads": EMPTY_STRING});
+        requestURI = string `${requestURI}${queryParamStr}`;
+        http:Response httpResponse = check self.amazonS3->post(requestURI, request);
+
+        xml XMLPayload = check httpResponse.getXmlPayload();
+        if httpResponse.statusCode == http:STATUS_OK {
+            return getUploadId(XMLPayload);
+        } else {
+            return error(XMLPayload.toString());
+        }
+    }
+    
+    # Completes a multipart upload by assembling previously uploaded parts.
+    #
+    # + objectName - The name of the object  
+    # + bucketName - The name of the bucket  
+    # + payload - The file content that needed to be added to the bucket  
+    # + uploadId - The upload ID of the multipart upload  
+    # + partNumber - The part number of the object  
+    # + uploadPartHeaders - Optional headers for the upload
+    # + return - An error on failure or else `()`
+    remote isolated function uploadPart(
+            @display {label: "Object Name"} string objectName,
+            @display {label: "Bucket Name"} string bucketName,
+            @display {label: "File Content"} string|xml|json|byte[]|stream<io:Block, io:Error?> payload,
+            @display {label: "Upload ID"} string uploadId,
+            @display {label: "Part Number"} int partNumber,
+            @display {label: "UploadPart Headers"} UploadPartHeaders? uploadPartHeaders = ())
+            returns CompletedPart|error {
+
+        if objectName == EMPTY_STRING {
+            return error(EMPTY_OBJECT_NAME_ERROR_MSG);
+        }
+        if bucketName == EMPTY_STRING {
+            return error(EMPTY_BUCKET_NAME_ERROR_MSG);
+        }
+        http:Request request = new;
+
+        string requestURI = string `/${bucketName}/${objectName}`;
+        string queryParamStr = string `?partNumber=${partNumber}&uploadId=${uploadId}`;
+        
+        map<string> queryParamsMap = {
+            "partNumber": partNumber.toString(),
+            "uploadId": uploadId
+        };
+
+        map<string> requestHeaders = setDefaultHeaders(self.amazonHost);
+
+        populateUploadPartHeaders(requestHeaders, uploadPartHeaders);
+
+        if payload is byte[] {
+            request.setBinaryPayload(payload);
+        } else if payload is stream<io:Block, io:Error?> {
+            request.setByteStream(payload);
+        } else {
+            request.setPayload(payload);
+        }
+
+        check generateSignature(self.accessKeyId, self.secretAccessKey, self.region, PUT, requestURI, UNSIGNED_PAYLOAD,
+            requestHeaders, request, queryParams = queryParamsMap);
```

Comment:
IMO variables should be declared close to where they are used. Hence, suggesting following change.
```suggestion
        map<string> requestHeaders = setDefaultHeaders(self.amazonHost);

        populateUploadPartHeaders(requestHeaders, uploadPartHeaders);

        if payload is byte[] {
            request.setBinaryPayload(payload);
        } else if payload is stream<io:Block, io:Error?> {
            request.setByteStream(payload);
        } else {
            request.setPayload(payload);
        }

        map<string> queryParamsMap = {
            "partNumber": partNumber.toString(),
            "uploadId": uploadId
        };
        check generateSignature(self.accessKeyId, self.secretAccessKey, self.region, PUT, requestURI, UNSIGNED_PAYLOAD,
            requestHeaders, request, queryParams = queryParamsMap);
```